### PR TITLE
add sawmill scene to convert logs to planks

### PR DESCRIPTION
**Pull request by EthanVieira**
image

### CODE REVIEW

File: src/targets/enemies/goblin.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { Enemy } from "./enemy.js";
 import NormalBones from "../../items/bones/normal-bones.js";
+import AirTalisman from "../../items/tools/talismans/air-talisman.js";
```
Comment (CurtisGreen) [inline src/targets/enemies/goblin.js:3]: Was this supposed to be included previously?

File: src/targets/sawmill-operator.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        const chat = this.scene.scene.get(CONSTANTS.SCENES.CHAT);
+
+        // See if logs are selected
+        const selectedIndex = inv.curSelectedItemIndex;
+        const selectedItem = inv.inventory[selectedIndex];
+
+        if (
+            selectedIndex < 0 ||
+            !selectedItem ||
+            !this.validMaterialsToCostMap.hasOwnProperty(selectedItem.name)
```
Comment (CurtisGreen) [inline src/targets/sawmill-operator.js]: any reason to use `hasOwnProperty` over bracket notation?

Ex: 
```js
!this.validMaterialsToCostMap[selectedItem.name]
```

**Review by CurtisGreen — COMMENTED** [review]
Saw this error message when the level loaded

`this.targets[this.currentTargetIndex].updateProgress is not a function`

File: src/targets/enemies/goblin.js (hunk 1 above)
Comment (EthanVieira) [reply src/targets/enemies/goblin.js:3]: Yep lol

**Comment by EthanVieira** [comment]
> Saw this error message when the level loaded
> 
> `this.targets[this.currentTargetIndex].updateProgress is not a function`

looks like an issue with autoclickers.. also affects other levels that require selecting something in inventory. I'll try to fix.

Thinking about it more.. maybe we don't want autoclickers to work when performing an action that uses a resource? It would auto-consume resources that the player may not want to be consumed. The alternative would be to have a on/off switch for them. Having it only work on targets that don't require resources would fit more into the theme of clanmates since clanmates can't actually help with most skilling. Thoughts?

**Comment by CurtisGreen** [comment]
> Thinking about it more.. maybe we don't want autoclickers to work when performing an action that uses a resource? It would auto-consume resources that the player may not want to be consumed. The alternative would be to have a on/off switch for them. Having it only work on targets that don't require resources would fit more into the theme of clanmates since clanmates can't actually help with most skilling. Thoughts?

I agree, I think it makes sense to exclude things kinds of things

**Review by CurtisGreen — APPROVED** [review]

**Pull request merged**